### ISSUE TITLE
training checkpoint - 5500 (1 hour on 3090) 

### ISSUE THREAD

**Issue by johndpope** [johndpope/SPEAK-hack#1]
![image](https://github.com/johndpope/SPEAK-hack/assets/289994/6d5d1182-99c0-4c9d-8b15-b0d60d4acb80)

![Screenshot from 2024-06-28 00-43-05](https://github.com/johndpope/SPEAK-hack/assets/289994/bf12743d-2632-443e-8737-9ef49e47f126)

i had to rework the generator to use less layers / and use 64 x 64 image resizing.

![Screenshot from 2024-06-28 00-45-17](https://github.com/johndpope/SPEAK-hack/assets/289994/d9886893-a9bf-4616-86d4-6430cdd69f4a)

**Comment by johndpope** [comment]
@fenghe12  / @JaLnYn / @ChenyangWang95   
this might actually work. 

In megaportaits I use custom resnet50 probably safer to switch that in because otherwise the model is going to just discard the updates???. I check in the morning.

**Comment by francqz31** [comment]
@johndpope is it just me or sonnet 3.5 machine learning code output is actually way more readable than opus ? feels like actual working code this time !!!

**Comment by johndpope** [comment]
something maybe not quite right. i train overnight. 
this is still  epoch 0 -


checkpoint-86500
![recon_step_86500](https://github.com/johndpope/SPEAK-hack/assets/289994/4dc47c8a-36ea-46bb-b4f7-caf2e4282c2b)

![Screenshot from 2024-06-28 07-49-49](https://github.com/johndpope/SPEAK-hack/assets/289994/79df090c-87ee-4f34-b9eb-46f7ddd8d441)

i change the code back to use 512x512
resume training - and get this.

![recon_step_87000](https://github.com/johndpope/SPEAK-hack/assets/289994/43e000be-ce41-4f9e-88eb-a15a25c148d3)

im seeing newer clearer images advancing in epoch 1 - even after a few more cycles - will udpate here later. i think by epoch 4 - probably going to be fairly decent. 


i add some tensorboard stuff - and surface the losses.

recon_step_126000.png
![image](https://github.com/johndpope/SPEAK-hack/assets/289994/407e128e-9164-428f-8aed-5a4813fbb1d0)


UPDATE - 
my bad was overfitting to one image. I just push updated dataloader.
new debug image.
![debug_step_164000](https://github.com/johndpope/SPEAK-hack/assets/289994/5f959004-10c9-4f0d-b2e2-d4030c1609a9)

Starting training again.
was seeing OOM errors - check your num_of_workers.
![debug_step_168000](https://github.com/johndpope/SPEAK-hack/assets/289994/8eda5d6c-89e9-49c9-9c64-1a264ca79263)


UPDATE - i restart training - I  change the generator to use resblocks - maybe will help recreate the image better.

![Screenshot from 2024-06-28 16-40-33](https://github.com/johndpope/SPEAK-hack/assets/289994/b4911298-ebde-4a49-9c4d-5fd1b3f0db98)


![debug_step_4000](https://github.com/johndpope/SPEAK-hack/assets/289994/bf40d223-f95b-4af7-b91b-5f8969f71106)



UPDATE - Sunday
so i rebuilt code to do progressive training with resolution upscaling - 64,128....256 ...512
added tensorboard losses
![Screenshot from 2024-06-30 05-43-40](https://github.com/johndpope/SPEAK-hack/assets/289994/aec76f4d-eab9-4c7a-8132-cc08ead89c0b)

i give up training across celebA - 
i overfit to one pair of images....

training progress so far
![Screenshot from 2024-06-30 14-10-09](https://github.com/johndpope/SPEAK-hack/assets/289994/e8d022f8-233f-489c-a10c-c6dcdc6f088b)


UPDATE - Sunday night

so had some battle with gradient explosions

ending up having to add some accummulation steps in that helped stablize things
https://github.com/johndpope/SPEAK-hack/pull/3

looks like the learning rate is getting things into a minima....
![debug_step_3500_resolution_64](https://github.com/johndpope/SPEAK-hack/assets/289994/0a9cfbbf-0adb-4241-a36c-b8e55cba18ea)



UPDATE - i switch to use 256 because resnet50 cant return rich features 2048,7,7 for images less than 224x224.

![Screenshot from 2024-07-01 09-50-51](https://github.com/johndpope/SPEAK-hack/assets/289994/2100a209-ce44-4e80-8e82-a877dee2ec22)
